provider/core/src/versions.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[derive(Debug, Clone, Default, Copy, PartialEq)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[non_exhaustive]
+/// Source data versions
+pub struct SourceVersions<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    /// The version of CLDR source data, if used during datagen.
+    pub cldr: Option<SourceVersion<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    /// The version of ICU source data, if used during datagen.
+    pub icuexport: Option<SourceVersion<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    /// The version of segmenter LSTM data, if used during datagen.
+    pub segmenter_lstm: Option<SourceVersion<'data>>,
+}
+
+impl SourceVersions<'_> {
+    /// Creates an empty struct.
+    pub const fn empty() -> Self {
+        SourceVersions {
+            cldr: None,
+            icuexport: None,
+            segmenter_lstm: None,
+        }
+    }
+}
+
+#[cfg(feature = "datagen")]
+impl databake::Bake for SourceVersions<'_> {
```

Comment:
thought: ideally would be very cool if e.g. the SourceVersions for properties did not mention a cldr version (since it's irrelevant)

but that seems to be more hassle than it's worth to track